### DISCUSSION TITLE
Problem with demo3

### DISCUSSION THREAD

**Discussion by shojaee1786** [InterDigitalInc/AdvantEDGE#363] [Q&A] [answered]
Hi there,
After clicking the "Register Application" button in the frontend, I encounter the following messages in the backend server:

INFO[2022-06-30T17:42:15.626Z] https://try-mec.etsi.org/sbx0hytdma/mep1/mec_app_support/v1  meep.component=Demo-3
DEBU[2022-06-30T17:42:15.948Z] Sending request to mec platform add app terminate subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1236"
DEBU[2022-06-30T17:42:16.052Z] Sending request to mec platform add service-avail subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1192"
INFO[2022-06-30T17:42:16.157Z] Subscribed to service availibility notification on mec platform  meep.component=Demo-3
DEBU[2022-06-30T17:42:16.266Z] Sending request to mec platform create ams resource api  meep.component=Demo-3 meep.from="demo3_service.go:991"
DEBU[2022-06-30T17:42:16.37Z] Sending request to mec platform adding ams subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1074"
2022/06/30 17:42:16 http: panic serving 5.115.160.38:33448: runtime error: invalid memory address or nil pointer dereference
goroutine 39 [running]:
net/http.(*conn).serve.func1(0xc0004d7720)
        /usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x811000, 0xb0f560)
        /usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.amsSendSubscription(0xc00001c180, 0x24, 0x0, 0x0, 0xc000018b40, 0x1a, 0x0, 0x0, 0xc0002e8f00, 0xc00035e410)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/demo3_service.go:1096 +0x2d1
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.demo3Register(0x906ca0, 0xc0004f7180, 0xc000132900)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/demo3_service.go:366 +0xf05
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.Register(0x906ca0, 0xc0004f7180, 0xc000132900)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/api_frontend.go:50 +0x3f
net/http.HandlerFunc.ServeHTTP(0x89cde8, 0x906ca0, 0xc0004f7180, 0xc000132900)
        /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.Logger.func1(0x906ca0, 0xc0004f7180, 0xc000132900)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/logger.go:35 +0xb0
net/http.HandlerFunc.ServeHTTP(0xc00017e5d0, 0x906ca0, 0xc0004f7180, 0xc000132900)
        /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00017c000, 0x906ca0, 0xc0004f7180, 0xc000132400)
        /root/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xd3
github.com/gorilla/handlers.(*cors).ServeHTTP(0xc0001781b0, 0x906ca0, 0xc0004f7180, 0xc000132400)
        /root/go/pkg/mod/github.com/gorilla/handlers@v1.5.1/cors.go:138 +0x2a8
net/http.serverHandler.ServeHTTP(0xc00013e0e0, 0x906ca0, 0xc0004f7180, 0xc000132400)
        /usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc0004d7720, 0x907a20, 0xc000416400)
        /usr/local/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2969 +0x36c
INFO[2022-06-30T17:42:16.677Z] https://try-mec.etsi.org/sbx0hytdma/mep1/mec_app_support/v1  meep.component=Demo-3
DEBU[2022-06-30T17:42:16.909Z] Sending request to mec platform add app terminate subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1236"
DEBU[2022-06-30T17:42:17.015Z] Sending request to mec platform add service-avail subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1192"
INFO[2022-06-30T17:42:17.121Z] Subscribed to service availibility notification on mec platform  meep.component=Demo-3
DEBU[2022-06-30T17:42:17.229Z] Sending request to mec platform create ams resource api  meep.component=Demo-3 meep.from="demo3_service.go:991"
DEBU[2022-06-30T17:42:17.334Z] Sending request to mec platform adding ams subscription api  meep.component=Demo-3 meep.from="demo3_service.go:1074"
2022/06/30 17:42:17 http: panic serving 5.115.160.38:33449: runtime error: invalid memory address or nil pointer dereference
goroutine 106 [running]:
net/http.(*conn).serve.func1(0xc00007f720)
        /usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x811000, 0xb0f560)
        /usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.amsSendSubscription(0xc00001c180, 0x24, 0x0, 0x0, 0xc000018b40, 0x1a, 0x0, 0x0, 0xc0002e84c0, 0xc000187f80)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/demo3_service.go:1096 +0x2d1
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.demo3Register(0x906ca0, 0xc0003d60e0, 0xc000221200)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/demo3_service.go:366 +0xf05
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.Register(0x906ca0, 0xc0003d60e0, 0xc000221200)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/api_frontend.go:50 +0x3f
net/http.HandlerFunc.ServeHTTP(0x89cde8, 0x906ca0, 0xc0003d60e0, 0xc000221200)
        /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/server.Logger.func1(0x906ca0, 0xc0003d60e0, 0xc000221200)
        /home/ubuntu/AdvantEDGE/examples/demo3/src/backend/server/logger.go:35 +0xb0
net/http.HandlerFunc.ServeHTTP(0xc00017e5d0, 0x906ca0, 0xc0003d60e0, 0xc000221200)
        /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00017c000, 0x906ca0, 0xc0003d60e0, 0xc0000fd100)
        /root/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xd3
github.com/gorilla/handlers.(*cors).ServeHTTP(0xc0001781b0, 0x906ca0, 0xc0003d60e0, 0xc0000fd100)
        /root/go/pkg/mod/github.com/gorilla/handlers@v1.5.1/cors.go:138 +0x2a8
net/http.serverHandler.ServeHTTP(0xc00013e0e0, 0x906ca0, 0xc0003d60e0, 0xc0000fd100)
        /usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc00007f720, 0x907a20, 0xc000219300)
        /usr/local/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2969 +0x36c

**Comment by roymx** [comment, marked as answer]
Hello @shojaee1786 ,

The MEC Sandbox has recently been updated by ETSI and is running a new version of AMS (v2.2.1)
> Note: ETSI forgot to update the version on the site landing page so it still shows 1.6,  but AMS version has changed

Demo3 is still using AMS v2.1.1 and this is what is causing the crash.

Unfortunately, I do not have an immediate workaround for this problem.
I will keep you informed via this thread as soon as there is a resolution.

**Reply by shojaee1786** [reply]
Dear Mike,

I appreciate your quick and comprehensive response.
Please, keep me informed of updates.

Kind regards,
Reza Shojaee
